api/observation.go

```diff
@@ -93,23 +95,23 @@ func (api *API) doGetObservations(ctx context.Context, datasetID, edition, versi
 		return nil, err
 	}
 
-	if versionDoc.CSVHeader == nil || versionDoc.Dimensions == nil {
+	if err = models.CheckState(models.Version, versionDoc.State); err != nil {
+		logData["state"] = versionDoc.State
+		log.Event(ctx, "get observations: unpublished version has an invalid state", log.ERROR, log.Error(err), logData)
+		return nil, err
+	}
+
+	if versionDoc.Dimensions == nil {
 		logData["version_doc"] = versionDoc
-		log.Event(ctx, "get observations", log.ERROR, log.Error(errs.ErrMissingVersionHeadersOrDimensions), logData)
-		return nil, errs.ErrMissingVersionHeadersOrDimensions
+		log.Event(ctx, "get observations", log.ERROR, log.Error(errs.ErrMissingVersionDimensions), logData)
```

Comment:
should there be some additional context in this log message? (maybe the error type provides enough context)